cmd/minikube/cmd/root.go

```diff
@@ -72,9 +73,26 @@ func Execute() {
 	}
 }
 
+// Handle config values for flags used in external packages (e.g. glog)
+// by setting them directly, using values from viper when not passed in as args
+func setFlagsUsingViper() {
+	setFlagValues := func(a *pflag.Flag) {
+		// If the flag is set, override viper value
+		if a.Changed {
+			viper.Set(a.Name, a.Value.String())
+		}
+		// Viper will give precedence first to calls to the Set command,
+		// then to values from the config.yml
+		a.Value.Set(viper.GetString(a.Name))
+	}
+
+	pflag.VisitAll(setFlagValues)
+}
+
 func init() {
-	RootCmd.PersistentFlags().BoolVarP(&showLibmachineLogs, "show-libmachine-logs", "", false, "Whether or not to show logs from libmachine.")
+	RootCmd.PersistentFlags().Bool(showLibmachineLogs, false, "Whether or not to show logs from libmachine.")
 	pflag.CommandLine.AddGoFlagSet(goflag.CommandLine)
+	viper.BindPFlags(RootCmd.PersistentFlags())
```

Comment:
I don't think so -- the persistent flags on RootCmd are only show-libmachine-logs, which we are using in the PersistentPreRun.  I've done some manual testing with `minikube start` but I'm not sure how I could mock those out in tests.